src/FileOperation/jpeg_file.cpp

```diff
@@ -0,0 +1,122 @@
+#include <stdlib.h>
+
+#ifdef _WIN32
+#else
+#include <jpeglib.h>
+#endif
+
+#include <setjmp>
+#include "jpeg_file.h"
+#include "../image_exception.h"
+
+namespace
+{
+    struct penguinv_err_mgr {
```

Comment:
Why do we need these structure and function in first place? Can't we just use our own C++ exception (`image_exception`)?

`setjmp` and `longjmp` are very dangerous and should not be used in C++ code

Reply:
We have the option of using the default jpeg library error handler. But overriding it will allow us to throw more meaningful errors. I can remove this, but it would be nice to have in the future.

Reply:
Sorry, missed your reply ealier. We better do not use `setjmp` and `longjmp` so default handler would be fine. We need to only know whether we succeed to create an image or we failed.

Reply:
Okay, looks like you are right and jpeg library indeed uses such technique.